nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/opvalidation/TransformOpValidation.java

```diff
@@ -2213,6 +2215,42 @@ public void testMergeAvgBp() {
 
     }
 
+    @Test
+    public void testReverseBp() {
+
+        Nd4j.getRandom().setSeed(12345);
+        SameDiff sd = SameDiff.create();
+        SDVariable input = sd.var(Nd4j.createFromArray(new double[][]{{2,7}, {3,5}, {4,5}}));
+        SDVariable out = new Reverse(sd, input,0,1).outputVariable();
+        out.markAsLoss();
+        String err = OpValidation.validate(new TestCase(sd)
+                .gradientCheck(true));
+        assertNull(err);
```

Comment:
When checking gradients, better to use stdev for the loss
The reason being is that (like here) a non-scalar output will be added (i.e., sum operation) to get a scalar loss.
The gradient backwards will be the same

i.e., dL/dout[i] is the same for all values of `i` whereas it's different for each `i` for `dL/d(std(out))`.
The reason this matters is that we can miss whole classes of errors with this sort of "gradient at all array positions is the same" type test.

Same thing for upsampling3d